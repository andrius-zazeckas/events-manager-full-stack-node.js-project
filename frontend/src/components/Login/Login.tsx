import { Box, Button, TextField } from "@mui/material";
import { type FC, FormEventHandler, useState } from "react";
import axios from "axios";
import { useLocation, useNavigate } from "react-router-dom";

export const Login: FC = () => {
  const [userData, setUserData] = useState({ username: "", password: "" });

  const navigate = useNavigate();

  const { pathname } = useLocation();

  const handleUserDataChange = (
    value: string,
    key: "username" | "password"
  ) => {
    setUserData((prevUserData) => ({ ...prevUserData, [key]: value }));
  };

  const resetForm = () => {
    setUserData({ username: "", password: "" });
  };

  const handleFormSubmit: FormEventHandler<HTMLFormElement> = (e) => {
    e.preventDefault();

    axios
      .post("http://localhost:5000/login", {
        username: userData.username,
        password: userData.password,
      })
      .then((res) => {
        alert(res.data.message);

        localStorage.setItem("token", res.data.token);

        document.cookie = `id=${res.data.id}`;

        resetForm();

        if (pathname === "/home") {
          window.location.reload();
        }

        navigate("/home");
      })
      .catch((error) => {
        alert(error.response.data.error);
      });
  };

  return (
    <form onSubmit={handleFormSubmit} aria-label="login page">
      <Box textAlign="center" my="30px">
        <Box>
          <TextField
            required
            label="login"
            placeholder="username"
            value={userData.username}
            onChange={(e) => handleUserDataChange(e.target.value, "username")}
          />
        </Box>
        <Box my="10px">
          <TextField
            required
            label="password"
            placeholder="password"
            type="password"
            value={userData.password}
            onChange={(e) => handleUserDataChange(e.target.value, "password")}
          />
        </Box>
        <Button type="submit" variant="outlined">
          Login
        </Button>
      </Box>
    </form>
  );
};
