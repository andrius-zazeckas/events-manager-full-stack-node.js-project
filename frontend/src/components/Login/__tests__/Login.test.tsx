import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { Login } from "../Login";

describe("Events", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
