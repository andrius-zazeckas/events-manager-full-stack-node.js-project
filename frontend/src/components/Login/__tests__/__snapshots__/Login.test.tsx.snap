// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events matches snapshot 1`] = `
<form
  aria-label="login page"
  onSubmit={[Function]}
>
  <div
    className="MuiBox-root css-du354x"
  >
    <div
      className="MuiBox-root css-0"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          login
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="username"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                login
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-1xo3mso"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r1:"
          id=":r1:-label"
        >
          password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r1:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="password"
            required={true}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                password
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Login
    </button>
  </div>
</form>
`;
